_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -14,20 +14,934 @@ class Jetpack_Core_API_Module_Activate_Endpoint
+					$updated = $grouped_options_current != $grouped_options ? update_option( 'stats_options', $grouped_options ) : true;
+					break;
+
+				case 'wp_mobile_featured_images':
+				case 'wp_mobile_excerpt':
+					$value = ( 'enabled' === $value ) ? '1' : '0';
+					// break intentionally omitted
+				default:
+					// If option value was the same, consider it done.
+					$updated = get_option( $option ) != $value ? update_option( $option, $value ) : true;
+					break;
+			}
+
+			// The option was not updated.
+			if ( ! $updated ) {
+				$not_updated[ $option ] = $error;
+			}
+		}
+
+		if ( empty( $invalid ) && empty( $not_updated ) ) {
+			// The option was updated.
+			return rest_ensure_response( $response );
+		} else {
+			$invalid_count = count( $invalid );
+			$not_updated_count = count( $not_updated );
+			$error = '';
+			if ( $invalid_count > 0 ) {
+				$error = sprintf(
+					/* Translators: the plural variable is a comma-separated list. Example: dog, cat, bird. */
+					_n( 'Invalid option for this module: %s.', 'Invalid options for this module: %s.', $invalid_count, 'jetpack' ),
+					join( ', ', $invalid )
+				);
+			}
+			if ( $not_updated_count > 0 ) {
+				$not_updated_messages = array();
+				foreach ( $not_updated as $not_updated_option => $not_updated_message ) {
+					if ( ! empty( $not_updated_message ) ) {
+						$not_updated_messages[] = sprintf(
+							/* Translators: the first variable is a module option name. The second is the error message . */
+							__( 'Extra info for %1$s: %2$s', 'jetpack' ),
+							$not_updated_option, $not_updated_message );
+					}
+				}
+				if ( ! empty( $error ) ) {
+					$error .= ' ';
+				}
+				$error .= sprintf(
+					/* Translators: the plural variable is a comma-separated list. Example: dog, cat, bird. */
+					_n( 'Option not updated: %s.', 'Options not updated: %s.', $not_updated_count, 'jetpack' ),
+					join( ', ', array_keys( $not_updated ) ) );
+				if ( ! empty( $not_updated_messages ) ) {
+					$error .= ' ' . join( '. ', $not_updated_messages );
+				}
+
+			}
+			// There was an error because some options were updated but others were invalid or failed to update.
+			return new WP_Error( 'some_updated', esc_html( $error ), array( 'status' => 400 ) );
+		}
+
+	}
+
+	/**
+	 * Calls WPCOM through authenticated request to create, regenerate or delete the Post by Email address.
+	 * @todo: When all settings are updated to use endpoints, move this to the Post by Email module and replace __process_ajax_proxy_request.
+	 *
+	 * @since 4.1.0
+	 *
+	 * @param string $endpoint Process to call on WPCOM to create, regenerate or delete the Post by Email address.
+	 * @param string $error	   Error message to return.
+	 *
+	 * @return array
+	 */
+	private function _process_post_by_email( $endpoint, $error ) {
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return array( 'message' => $error );
+		}
+
+		$this->xmlrpc->query( $endpoint );
+
+		if ( $this->xmlrpc->isError() ) {
```

Comment:
This `xmlrpc` property is private in the parent class so it's inaccessible from the children.